app/script/conversation/ConversationEphemeralHandler.js

```diff
@@ -226,6 +197,16 @@ z.conversation.ConversationEphemeralHandler = class ConversationEphemeralHandler
     this.logger.info(`Obfuscated text message '${messageEntity.id}'`);
   }
 
+  _timeoutEphemeralMessage(messageEntity) {
+    if (!messageEntity.is_expired()) {
+      if (messageEntity.user().is_me) {
+        return this._obfuscateMessage(messageEntity);
+      }
+
+      amplify.publish(z.event.WebApp.CONVERSATION.EPHEMERAL_MESSAGE_TIMEOUT, messageEntity);
```

Comment:
I would suggest some kind of convention that limits the usage of event dispatchers in leave components/handlers/...

Since the repository is the interface with the external world (mostly views), I would say that it is also the one responsible for warning the external world that something happened (in that case, firing an event).

Therefor, I would suggest to restore the explicit callback passing to the `handler`. If you find the API confusing (passing the callback in the constructor), I believe we can find something better.
Another idea would be 
`handler.onMessageTimeout(this.handleMessageTimeout)` 
for example.

WDYT?

Reply:
The ideal scenario (in my opinion) would be that handlers are local event emitters (not responsible for side effects) and that a parent component could listen to the events it is interested in.

Reply:
Look at the previous code though. We called ourselves...

Reply:
Not sure I understand your comment?

Reply:
you mean the `this.onMessageTimeout`? This is actually a function that comes from the `ConversationRepo`, it is given in the constructor